templates/collection.yaml

```diff
@@ -519,15 +519,26 @@ Resources:
   LambdaLogGroup:
     Type: 'AWS::Logs::LogGroup'
     Properties:
-      LogGroupName: !Sub '/aws/lambda/${Lambda}'
+      LogGroupName: !Sub
+        - '/aws/lambda/${LambdaName}'
+        - LambdaName: !If
+            - LambdaSetToStackNameIsEnabled
+            - !Ref 'AWS::StackName'
+            - !Join
+                - ''
+                - - !Select [ 0, !Split [ 'LambdaRole', !Sub '${LambdaRole}' ] ]
+                  - 'Lambda'
+                  - !Select [ 1, !Split [ 'LambdaRole', !Sub '${LambdaRole}' ] ]
       RetentionInDays: !Ref LogGroupExpirationInDays
   Lambda:
     Type: 'AWS::Lambda::Function'
+    DependsOn:
+        - LambdaLogGroup
```

Comment:
I can't tell if this is necessary given you explicitly depend on lambda log group below (albeit for one of the conditional branches). I'm fine with keeping as is to make sure the dependency isn't optimized away if stack name is used.

Reply:
I also want it there in git blame so it points to this commit if anyone thinks of removing it again. Maybe I'm paranoid